src/main/java/seedu/expense/logic/commands/AddCommand.java

```diff
@@ -55,7 +55,7 @@ public CommandResult execute(Model model) throws CommandException {
         if (!model.hasCategory(toAdd.getTag())) {
             Expense expense = toAdd.resetTag();
             model.addExpense(expense);
-            return new CommandResult(String.format(MESSAGE_DEFAULT, expense));
+            return new CommandResult(String.format(MESSAGE_DEFAULT, toAdd.getTag().tagName));
```

Comment:
Will this work even after the tag has been reset?

Reply:
Yes, this works because `Expense` is immutable, and `Expense#resetTag` returns a new `Expense` object instead of modifying the original object.